builtins/amp-img.js

```diff
@@ -178,6 +203,21 @@ export class AmpImg extends BaseElement {
     return promise;
   }
 
+  /**
+   * Sets the img src to the first url in the srcset if srcset is defined but
+   * src is not.
+   * @private
+   */
+  guaranteeSrcForSrcsetUnsupportedBrowsers_() {
+    if (!this.img_.hasAttribute('src') && 'srcset' in this.img_ == false) {
+      const srcset = this.element.getAttribute('srcset');
+      const srcseturl = srcsetRegex.exec(srcset);
+      if (srcseturl) {
+        this.img_.setAttribute('src', srcseturl[0]);
```

Comment:
`match` would return an array of arrays. So [0] would be an array itself. First item should the the whole line match (e.g. `file.png 200w`) [1] should be the first grouping which is the URL and what we want. let's add a test for it as well.

Reply:
It uses `RegExp#exec`, not `match`. 😉

But, you do want the url at `[1]`, not the entire source at `[0]`.

Reply:
Good catch!

Reply:
So if we think about this, we really only need the first url in the `srcset`. It's a url that can be `http`, `https` or relative, and ends with a space. So can we just match the first `\S+` and be done with it? If we wanted to validate more, we could do `(http|https|\/)\S+`.